File: monitorlibrary/lib/data/section.dart
import 'package:meta/meta.dart';
import 'package:monitorlibrary/data/question.dart';

import 'content.dart';

class Section {
  String? title, description, sectionId;
  List<Question>? questions;
  String? sectionNumber;

  Section(
      {required this.title,
        required this.sectionNumber,
        required this.questions, required this.sectionId,
        required this.description});

  Section.fromJson(Map data) {
    this.sectionId = data['sectionId'];
    this.title = data['title'];
    this.sectionNumber = data['sectionNumber'];
    this.questions = [];
    if (data['questions'] != null) {
      List  list = data['questions'];
      list.forEach((m) {
        this.questions!.add(Question.fromJson(m));
      });
    }
    this.description = data['description'];

  }
  Map<String, dynamic> toJson() {
    List mQuestions = [];
    if (questions != null) {
      questions!.forEach((photo) {
        mQuestions.add(photo.toJson());
      });
    }

    Map<String, dynamic> map = {
      'title': title,
      'sectionNumber': sectionNumber,
      'description': description,
      'questions': mQuestions,
      'sectionId': sectionId,


    };
    return map;
  }
}

File: backend/src/main/java/com/monitor/backend/data/Section.java
package com.monitor.backend.data;

import java.util.List;

public class Section {
    private int  sectionNumber;
    private String  title, sectionId;
    private String  description;
    private List<Question> questions;

    public Section() {
    }

    public Section(String sectionId, int sectionNumber, String title, String description, List<Question> questions) {
        this.sectionNumber = sectionNumber;
        this.title = title;
        this.description = description;
        this.questions = questions;
    }

    public String getSectionId() {
        return sectionId;
    }

    public void setSectionId(String sectionId) {
        this.sectionId = sectionId;
    }

    public int getSectionNumber() {
        return sectionNumber;
    }

    public void setSectionNumber(int sectionNumber) {
        this.sectionNumber = sectionNumber;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<Question> getQuestions() {
        return questions;
    }

    public void setQuestions(List<Question> questions) {
        this.questions = questions;
    }
}
